### DISCUSSION TITLE
Connect via USB

### DISCUSSION THREAD

**Discussion by Druzecode** [tjko/nxgipd#18] [General]
Do you know what cable is needed to connect the Caddx serial interface to a USB port?  Does it just need a standard null modem cable?  Something like this: https://www.amazon.com/StarTech-com-Modem-Serial-Adapter-ICUSB232FTN/dp/B008634VJY/ref=sr_1_1?ie=UTF8&qid=1473133103&sr=8-1&keywords=null+modem+to+usb

Thanks,
Doc

**Comment by Dogora** [comment]
Pretty much any USB-to-RS232 adapter will work, including the one you list from Amazon. The NX-584 interface board has jumpers on it to swap pins as necessary to act like either DCE or DTE. The RS-232 cable between the panel and computer can be straight-through for convenience. You won't hurt anything with it hooked up wrong, so just try the jumpers in different positions until it works.
Due to its much higher speed, USB has rather more stringent length restrictions that RS-232. At 9600 baud, RS-232 can go easily 50 feet. My panel is at least 30 feet from the monitoring computer, yet is quite reliable at 9600 baud.

**Comment by Druzecode** [comment]
Thanks for the response.  I was planning to use a small system near the alarm system to run this, so distance isn't a big concern.  Also, I don't have an NX-584 board, just the NX-8E board.  I figured I could use one of these to attach: http://www.homecontrols.com/Interlogix-NetworX-Serial-Cable-GEP0003

Hopefully I can find one for cheaper, though.  I'm not super familiar with these systems.  Can I use that serial interface on the NX-8E, or so I need the NX-584 board?

Thanks,
Doc

**Comment by Dogora** [comment]
Use the serial port on the NX-8E. Pin-outs for that connector are in the Wiki for this project.
You can probably build your own cable cheaper, if you're handy with such things. Possibly, some cable for inside a computer case will work, but you'll have to check carefully. A typical motherboard probably has a 10-pin header for a serial port, if it has one.
You'll also need to program your NX-8E to communicate over the Home Automation system, in locations 208-211. Details on what to do are in the README for this project.
